packages/maker.js/src/core/model.ts

```diff
@@ -111,17 +111,17 @@ namespace MakerJs.model {
      * @returns Array of Caption objects.
      */
     export function getAllCaptionsOffset(modelContext: IModel) {
-        const captions: ICaption[] = [];
+        const captions: (ICaption & { layer?: string })[] = [];
 
-        function tryAddCaption(m: IModel, offset: IPoint) {
+        function tryAddCaption(m: IModel, offset: IPoint, layer: string) {
             if (m.caption) {
-                captions.push({ text: m.caption.text, anchor: path.clone(m.caption.anchor, offset) as IPathLine });
+                captions.push({ text: m.caption.text, anchor: path.clone(m.caption.anchor, offset) as IPathLine, layer });
```

Comment:
I think we need to capture the anchor here, and use its layer or the layer parameter. Since this function is called centrally from all exporter methods.